# license: modifies FCEUmm (https://github.com/libretro/libretro-fceumm), GPL-2.0 (Copying); this patch is distributed under the same license.
diff --git forkSrcPrefix/Makefile.libretro forkDstPrefix/Makefile.libretro
index 5b90d40..36e8033 100644
--- forkSrcPrefix/Makefile.libretro
+++ forkDstPrefix/Makefile.libretro
@@ -460,6 +460,22 @@ else ifeq ($(platform), miyoo)
 	SHARED := -shared -Wl,--version-script=src/drivers/libretro/link.T -Wl,-no-undefined
 	PLATFORM_DEFINES += -fomit-frame-pointer -ffast-math -mcpu=arm926ej-s
 
+# target platform tg5050 trimUI smart pro s
+else ifeq ($(platform), tg5050)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	SHARED := -shared -Wl,--version-script=src/drivers/libretro/link.T -Wl,-no-undefined
+	LDFLAGS += -fPIC -flto
+	CFLAGS += -mcpu=cortex-a55
+	CFLAGS += -fomit-frame-pointer -ffast-math -fPIC -flto
+	OPTIMIZE := -Ofast -DNDEBUG
+	HAVE_NTSC = 0
+	ifeq (,$(DEBUG))
+		# LDFLAGS += -s
+	endif
+	
 # Windows MSVC 2017 all architectures
 else ifneq (,$(findstring windows_msvc2017,$(platform)))
 	CC  = cl.exe
